Libraries/LibWeb/Painting/StackingContext.cpp

```diff
@@ -139,6 +139,19 @@ void StackingContext::paint_descendants(PaintContext& context, Paintable const&
         if (child.is_positioned() && z_index.value_or(0) == 0)
             return IterationDecision::Continue;
 
+        // https://www.w3.org/TR/CSS22/zindex.html
+        // All non-positioned floating descendants, in tree order. For each one of these, treat the
+        // element as if it created a new stacking context, but any positioned descendants and
+        // descendants which actually create a new stacking context should be considered part of the
+        // parent stacking context, not this new one.
+        auto floating_item_should_be_treated_as_stacking_context = child.is_floating() && !z_index.has_value();
```

Comment:
Can we use `!child.is_positioned()` to better reflect the spec?

Reply:
I originally left it out because positioned elements were already skipped the other condition. But this way it might read more easily.

I've also moved the condition that skips positioned elements below the new code now. Since the new code no longer relies on that condition, it makes more sense to keep the blocks that have a similar "code shape" directly next to each other.